ChallengeUI/src/main/java/com/drivequant/drivekit/challenge/ui/challengedetail/viewmodel/ChallengeDetailViewModel.kt

```diff
@@ -124,130 +113,40 @@ class ChallengeDetailViewModel(private val challengeId: String) : ViewModel() {
         }
     }
 
-    fun getBestPerformance(context: Context): String {
+    fun getBestPerformance(): String {
         return challengeDetailData?.let {
-            return when (challenge.themeCode) {
-                in 101..221,
-                401 -> {
-                    val score = if (it.challengeStats.maxScore == 10.0) {
-                        it.challengeStats.maxScore.removeZeroDecimal()
-                    } else {
-                        it.challengeStats.maxScore.format(2)
-                    }
-                    "$score/10"
-                }
-                in 306..309 ->formatChallengeDuration(it.challengeStats.maxScore, context).convertToString()
-                in 302..305 -> formatChallengeDistance(it.challengeStats.maxScore, context).convertToString()
-                301 -> "${it.challengeStats.maxScore.removeZeroDecimal()} ${context.resources.getQuantityString(
-                    com.drivequant.drivekit.common.ui.R.plurals.trip_plural,
-                    it.challengeStats.numberTrip
-                )}"
-                else -> "-"
+            val score = if (it.challengeStats.maxScore == 10.0) {
+                it.challengeStats.maxScore.removeZeroDecimal()
+            } else {
+                it.challengeStats.maxScore.format(2)
```

Comment:
Ce serait mieux d'appeler directement `it.challengeStats.maxScore.removeZeroDecimal().format(2)`, car si maxScore vaut `9.0` ou `8.0`, l'UI affiche "9.0/10" et "8.0/10" (idem pour les autres méthodes dessous)

Reply:
Je ne comprends pas trop. `format(2)` appelle `removeZeroDecimal(2)` et "9.0" est bien transformé en "9".
Si je fais ce que tu dis, "7.99" va devenir "8" au lieu de rester "7.99", et "4.56" va devenir "4.6".

En fait, j'ai l'impression qu'on peut plutôt appeler `it.challengeStats.maxScore.format(2)` dans tous les cas.

Reply:
Oui, tu as raison. Dans le cas du `7.99`, ça causerait problème. Le mieux serait donc d'appeler `format(2)` directement. A vérifier sur iOS que ce soit la même logique

Reply:
C'est fait différemment sur iOS mais c'est OK